lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -360,13 +360,18 @@ public function getCreatePrimaryKeySQL(Index $index, $table)
      */
     protected function getCreateColumnCommentSQL($tableName, $columnName, $comment)
     {
+        $schemaSQL = 'dbo';
+        $tableSQL  = $tableName;
         if (strpos($tableName, '.') !== false) {
             [$schemaSQL, $tableSQL] = explode('.', $tableName);
-            $schemaSQL              = $this->quoteStringLiteral($schemaSQL);
-            $tableSQL               = $this->quoteStringLiteral($tableSQL);
-        } else {
-            $schemaSQL = "'dbo'";
-            $tableSQL  = $this->quoteStringLiteral($tableName);
+        }
+
+        if (strpos($tableName, '[') === false) {
```

Comment:
I think we should `preg_replace('~^\[|\]$~s', '', $tableName)` instead and quote as literal as we would normally do.

String literal seeems to be the official way to escape: https://docs.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sp-addextendedproperty-transact-sql?view=sql-server-ver15#b-adding-an-extended-property-to-a-column-in-a-table

Reply:
> @metaturso please see #4360 , I think we need to fix in a little more abstract way - please provide feedback, if you want to change anything or help with the tests, please send a PR directly to my branch. Michael

@mvorisek thanks for stepping in with a patch for this, chap. Your fix in PR #4360 is reasonably self-contained and I'm happy to ditch my changes in favour of yours. 

There's definitely something about the amount of (un)escaping that's done at multiple levels that doesn't completely sit right with me, but fixing the data-flow is a much more pervasive change that's likely to break B/C or third-party code so I can't offer a better alternative.

I didn't get a chance to sit down and write the promised integration test. I'll commit and push the fixed version of the unit test, which runs against 2.11.x version of the codebase, during my lunch break. Feel free to steal this from my PR if that's of any use to you.

Should you decide not to use my test, please don't forget to cover for #4283 by adding test cases or fixture using i) table names with a custom schema name and ii) containing special characters or reserved keywords to your pull request.